components/homme/src/theta-l/prim_state_mod.F90

```diff
@@ -350,19 +349,16 @@ subroutine prim_printstate(elem, tl,hybrid,hvcoord,nets,nete)
     ! so rate of changes are W/m**2
 
 
-    !   mass = integral( ps-p(top) )
+    !   mass = integral( ps-p(top) ) = sum(dp3d)
     do ie=nets,nete
-       !tmp(:,:,ie)=elem(ie)%state%ps_v(:,:,n0) 
-       tmp(:,:,ie)=hvcoord%hyai(1)*hvcoord%ps0 + sum(elem(ie)%state%dp3d(:,:,:,n0),3) 
+       tmp(:,:,ie)=sum(elem(ie)%state%dp3d(:,:,:,n0),3) 
+#ifdef CAM
+       ! CAM neglicts the small, constant p(top) term.  Add this
+       ! term in to be consistent
+       tmp(:,:,ie)=temp(:,:,ie) + hvcoord%hyai(1)*hvcoord%ps0 
```

Comment:
If CAM ignores A(1)*p0 term, why is it added here in ifdefCAM section? If CAM's tmp=sum(dp3d) does not have this term, then maybe this ifdef should be ifndef?

Reply:
CAM does not ignore the A(1)*p0 term.  instead CAM neglicts the fact that p(top)>0 when computing the mass.  Adding this constant to the mass will make it consistent with the formula used by CAM.